x-pack/solutions/observability/plugins/infra/server/lib/host_details/process_list.ts

```diff
@@ -128,35 +131,280 @@ export const getProcessList = async (
+          by_status: {
+            terms: {
+              field: 'attributes.status',
+              size: 20,
+            },
+            aggs: {
+              latest_count: {
+                top_hits: {
+                  size: 1,
+                  sort: [
+                    {
+                      [TIMESTAMP_FIELD]: {
+                        order: 'desc',
+                      },
+                    },
+                  ],
+                  fields: ['system.processes.count'],
+                },
+              },
+            },
+          },
+        },
+      },
+      processes: {
+        filter: {
+          bool: {
+            must: [...filter, ...mandatoryFieldsFilter],
+          },
+        },
+        aggs: {
+          filteredProcs: {
+            terms: {
+              field: PROCESS_COMMANDLINE_FIELD,
+              size: TOP_N,
+              order: {
+                [sortBy.name]: sortBy.isAscending ? 'asc' : 'desc',
+              },
+            },
+            aggs: {
+              cpu: {
+                avg: {
+                  field: 'process.cpu.time',
+                },
+              },
+              memory: {
+                avg: {
+                  field: 'process.memory.utilization',
+                },
+              },
+              startTime: {
+                max: {
+                  field: 'start_timestamp',
+                },
+              },
+              meta: {
+                top_hits: {
+                  size: 1,
+                  sort: [
+                    {
+                      [TIMESTAMP_FIELD]: {
+                        order: 'desc',
+                      },
+                    },
+                  ],
+                  fields: ['process.pid', 'process.command_line', 'process.owner'],
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  };
+
+  const result = await search<{}, ProcessListAPIQueryAggregationSEMCONV>({
+    body,
+    index: sourceConfiguration.metricAlias,
+  });
+
+  const { buckets: processListBuckets } = result.aggregations!.processes.filteredProcs;
+  const processList = processListBuckets.map((bucket) => {
+    const hit = bucket.meta.hits.hits[0];
+    const meta = hit.fields || {};
+
+    const getValue = (fieldName: string) => {
+      const value = meta[fieldName];
+      return Array.isArray(value) ? value[0] : value;
+    };
+
+    const metricTimestamp = hit.fields?.['@timestamp']
+      ? new Date(getValue('@timestamp')).getTime()
+      : Date.now();
+
+    // Get start time from aggregation or fallback to current time
+    const startTime = bucket.startTime?.value_as_string
+      ? Date.parse(bucket.startTime.value_as_string)
+      : metricTimestamp;
+
+    const processRuntimeMs = metricTimestamp - startTime;
+
+    const cpuTimeValueMs = bucket.cpu.value * 1000; // convert seconds to milliseconds
+    const cpu = processRuntimeMs > 0 ? cpuTimeValueMs / processRuntimeMs : 0;
```

Comment:
If we use the `process.cpu.utilization` field I believe we can skip everything here

Reply:
based on this remapper file: https://github.com/smith/opentelemetry-lib/blob/cd0e3efbcd9210a516f500896e4e1a148ea8522a/remappers/hostmetrics/process.go#L261

Reply:
Do you think I should use `process.cpu.utilization`?